docs/general/naming.md

```diff
@@ -0,0 +1,312 @@
+particular programming language or wire format._
+
+Names SHOULD follow these rules:
+
+- Names SHOULD be lowercase.
+
+- Use namespacing. Delimit the namespaces using a dot character.
+  For example `service.version` denotes the service version where
+  `service` is the namespace and `version` is an attribute in that namespace.
+
+- Namespaces can be nested. For example `telemetry.sdk` is a namespace inside
+  top-level `telemetry` namespace and `telemetry.sdk.name` is an attribute
+  inside `telemetry.sdk` namespace.
+
+  Use namespaces (and dot separator) whenever it makes sense. For example
+  when introducing an attribute representing a property of some object, follow
+  `*{object}.{property}` pattern. Avoid using underscore (`*{object}_{property}`)
+  if this object could have other properties.
+
+- For each multi-word dot-delimited component of the name separate the
+  words by underscores (i.e. use snake_case). For example
+  `http.response.status_code` denotes the status code in the http namespace.
+
+  Use underscore only when using dot (namespacing) does not make sense or changes
+  the semantical meaning of the name. For example, use `rate_limiting`
+  instead of `rate.limiting`.
+
+- Be precise. Attribute, event, metric, and other names should be descriptive and
+  unambiguous.
+  - When introducing a name describing certain property of the object,
+    include the property name. For example, use `file.owner.name` instead of `file.owner`
+    and `system.network.packets.dropped` instead of `system.network.dropped`
+  - Avoid introducing names and namespaces that would mean different things when
+    used by different conventions or instrumentations. For example, use `security_rule`
+    instead of `rule`.
+
+- Use shorter names when it does not compromise clarity. Drop namespace
+  components or words in multi-word components when they are not necessary. For example,
+  `vcs.change.id` describes pull request id as precisely as `vcs.repository.change.id` does.
+
+## Name Abbreviation Guidelines
+
+Abbreviations MAY be used when they are widely recognized and commonly used.
+
+Examples include common technical abbreviations such as `IP`, `DB`, `CPU`,
+`HTTP`, `URL`, or product names like `AWS`, `GCP`, `K8s`.
+
+Abbreviations that are commonly recognized but only within a certain domain MAY
+be used when qualified by the corresponding namespace.
+
+For example, use `container.csi.*` instead of `container.container_storage_interface`
+or `container.oci.*` instead of `container.open_container_initiative.*`
+
+Abbreviations SHOULD be avoided if they are ambiguous, for example, when they apply
+to multiple products or concepts.
+
+## Name Reuse Prohibition
+
+Two attributes, two metrics, or two events MUST NOT share the same name.
+Different entities (attribute and metric, metric and event) MAY share the same name.
+
+Attributes, metric, events, or other named entities SHOULD NOT be removed from semantic
```

Comment:
what are other named entities that this applies to?

Reply:
changed  to 

> Attributes, metrics, and events SHOULD NOT be removed from semantic

I was thinking that we should apply this doc to all named things including future ones (which might include resources/entities, profiles, span type, etc). 
Let's stop at current named things for now.